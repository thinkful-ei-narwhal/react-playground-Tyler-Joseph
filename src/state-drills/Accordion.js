import React from 'react';


export default class Accordion extends React.Component {
  store = [...this.props.sections]
  
  state = {
    lastClickedButton: null
  }

  handleClick = (index) => {
    this.setState({
      lastClickedButton: index
    })
  }


  actions = (data, index, lastClickedButton) => {
    return ( 
        <li key={index}>
          <button name={index} onClick={() => this.handleClick(index)}>{data.title}</button>
          {(lastClickedButton === index) && <p>{data.content}</p>}
        </li>
    )
  }

  render() {
    const { lastClickedButton } = this.state
    return (
      <ul>
      {this.store.map((x, i) => this.actions(x, i, lastClickedButton))}
      </ul>
    )
  }
}