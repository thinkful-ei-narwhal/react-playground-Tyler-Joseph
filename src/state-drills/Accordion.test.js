import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import toJson from 'enzyme-to-json';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe('<Accordion />', () => {
  it('renders empty \'li\' if sections not supplied', () => {
    const items = []
    const tree = renderer.create(<Accordion sections={items}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('component renders no section as active by default', () => {
    const tree = renderer.create(<Accordion sections={sections}/>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('opens any clicked section', () => {
    const wrapper = Enzyme.shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens one section at a time', () => {
    const wrapper = Enzyme.shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})})