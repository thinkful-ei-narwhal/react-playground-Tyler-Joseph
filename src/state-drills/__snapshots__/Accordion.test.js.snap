// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> component renders no section as active by default 1`] = `
<ul>
  <li>
    <button
      name={0}
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li>
    <button
      name={1}
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li>
    <button
      name={2}
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`<Accordion /> renders empty 'li' if sections not supplied 1`] = `<ul />`;
